### DISCUSSION TITLE
Landsat Appears to have Missing Scenes between the PC and USGS

### DISCUSSION THREAD

**Discussion by alexgleith** [microsoft/PlanetaryComputer#245] [General]
Hey folks

I've been experimenting with the Planetary Computer and have found a few examples where there are missing scenes, and within scenes, missing files.

Here's a notebook that demonstrates a discrepancy of 30 scenes over fiji: https://gist.github.com/alexgleith/8ca78959c03d4ff943dd4888432f84e4

And this scene has missing .tif files, or perhaps, the file paths are incorrect: https://planetarycomputer.microsoft.com/api/stac/v1/collections/landsat-c2-l2/items/LC08_L2SR_074073_20221105_02_T1 

I'd be interested to know if anyone has any ideas on why this is happening or if there are plans to correct any issues that may be causing the discrepancy?

**Comment by TomAugspurger** [comment]
Thanks for the report. We've had some trouble with getting a 100% robust pipeline working with USGS's servers.

The current version of the pipeline *should* be robust. It looks like the most recent missing scene from your list is from October 2022. For the missing scenes, I plan to start a job to backfill the archive sometime later this year, but haven't been able to get to it yet.

> And this scene has missing .tif files, or perhaps, the file paths are incorrect: https://planetarycomputer.microsoft.com/api/stac/v1/collections/landsat-c2-l2/items/LC08_L2SR_074073_20221105_02_T1

Yet another issue with our old pipeline, we'd get a successful HTTP response and upload the data, when in fact it was an HTML error page. Follow https://github.com/microsoft/PlanetaryComputer/discussions/101 for updates on that. We're working on it now.

**Reply by TomAugspurger** [reply]
Quick status update here: we've got a list of scenes with the bad TIFF files. USGS has reprocessed many of those scenes (and deleted the original data). I'm able to get a list of the matching scenes and will begin to copy the data over once USGS's distribution service has stabilized a bit. It will take some time for all of the bytes to be copied.

The STAC item IDs will not change (we don't include the processing timestamp in our item IDs). The assets for that scene will be all new, and the original assets will be deleted.

**Reply by alexgleith** [reply]
Sounds good!

For DE Africa, we noticed these reprocessed scenes too. We subscribed to their notifications and I believe we were able to keep up with the changes using that method. I guess that's what you're doing?

**Reply by TomAugspurger** [reply]
> We subscribed to their notifications and I believe we were able to keep up with the changes using that method.

I have a prototype that does that, but it isn't deployed yet. We're currently polling their API at regular intervals to find new scenes.

**Reply by alexgleith** [reply]
>polling their API at regular intervals

Polling by date range? Reprocessing obviously won't be identified this way, unfortunately.

**Reply by TomAugspurger** [reply]
By publishing date, so we do pick up reprocessed scenes.

**Reply by alexgleith** [reply]
>By publishing date

Oh cool :-)

**Comment by alexgleith** [comment]
Hey @TomAugspurger I re-ran my comparison over multiple years over Fiji, and it looks like the issue is only there for recent scenes, i.e., 2022 and some 2023. So this issue is less of a big deal than I thought it might be. Still, those missing scenes would be nice to have!

Notebook: https://gist.github.com/alexgleith/8ca78959c03d4ff943dd4888432f84e4

![image](https://github.com/microsoft/PlanetaryComputer/assets/3445853/ee4b268c-a4be-4b49-b764-2d685959700b)
